themes/default/content/blog/iac-recommended-practices-tying-stacks-together/index.md

```diff
@@ -0,0 +1,87 @@
+
+## Linking stacks
+
+For those that are unfamiliar, stack references are a feature in Pulumi that allows you to access output values from one stack in another stack. In other words, stack references provide a way to programmatically "link" stacks in different projects for the purpose of sharing information. You can find more details on stack references [here in the Pulumi docs](/docs/intro/concepts/stack/#stackreferences).
+
+Using stack references, the Zephyr team could make the necessary connections they needed:
+
+* The code handling the Kubernetes cluster could reference the VPC ID and subnet IDs from the base infrastructure layer.
+* Similarly, the code that was responsible for deploying the online store services could reference the necessary Kubernetes connection details for the Kubernetes cluster.
+
+It's worthwhile to note that any information that needs to be accessible from another stack via a stack reference must be exported as a stack output in the source stack. If you don't mark it as a stack output, then it can't be used in a stack reference. Fortunately, adding stack outputs after the fact is easy and generally has no impact whatsoever on existing infrastructure.
+
+Additionally, any value retrieved via a stack reference is treated as an `Output`, and therefore may require some additional work to transform values (such as the use of `Output.apply`). The recent addition of `OutputDetails` support in Pulumi---you can read more about that [here](/blog/stack-reference-output-details/)---helps enormously in this situation. Some SDKs also have language-specific mechanisms that can help; for example, using Go's `.AsStringArrayOutput()` method on a `StackReference.GetOutput` statement makes referencing subnet IDs from other project much easier.
+
+While stack references are generally straightforward, there are some recommended practices to observe around the use of stack outputs and stack references:
+
+1. With regard to stack outputs, **only export what is needed.** If there's a need for it to be accessed from outside the stack, export it; otherwise, don't. It's easy to add more stack outputs after the fact, and as has been mentioned already this is done with a quick `pulumi up` that has no affect on existing infrastructure (you're only modifying the stack object itself).
+2. **Be judicious in the use of stack references.** Each stack reference is a call to the backend to read a value, which adds some level of latency. Using lots of stack references (think more than 20 or so) can introduce notable latency in Pulumi operations.
```

Comment:
@ringods @pierskarsenbarg Since you folks brought up the concern regarding minimize stack references and calls to read values, can you confirm if the concern is in stack outputs or stack references? Based on the comments in Slack, I believe it is the reading stack (and therefore stack references are the concern).

Reply:
Resolved via Slack discussions (see threads in `#project-zephyr` and `#questions`).